core/jobs/src/main/java/com/cognifide/aet/job/common/utils/ParamsHelper.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Automated Exploratory Tests
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.aet.job.common.utils;
+
+
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.apache.commons.lang3.ObjectUtils;
+
+public class ParamsHelper {
```

Comment:
I like idea of this helper very much. It looks that lot of code is now simpler and redundancy is limited.
Please consider enriching this class's methods with javadoc, I think that some of this code may be reused in many future modifiers.

I'm also wondering if following scenarios:

1. Don't return `null`, use `Optional` instead.
2. Don't return `null`, provide option with 3rd - `default` parameter and return it when no param with key provided.